captum/attr/_core/integrated_gradients.py

```diff
@@ -53,6 +53,7 @@ def __init__(self, forward_func: Callable) -> None:
                           modification of it
         """
         GradientAttribution.__init__(self, forward_func)
+        self.predefined_step_size_alphas = None  # used for internal batching
```

Comment:
I think this can be dangerous if we call attribute multiple times for different inputs. Does it have to be an instance variable ?

Reply:
This attribute is only modified in `_batched_attribution` for recursive sub-calls and always reset to None after each operation, so it shouldn't affect any separate attribution calls. The other option is to add another argument to attribute that is only used for this, but it could be confusing to users looking at the method signature, so I preferred this approach.

Reply:
I think that if this gets executed in a multi-threaded environment there can be some issues... 

One way to avoid it could be to move the core logic of ig into an auxiliary function such as: _attribute that also takes start and ends kwargs. `_batched_attribution ` can call  `_attribute` instead of `attribute` ?
https://github.com/pytorch/captum/pull/333/files#diff-67ff19e8dcc2510648379db71c2ee9fbR275

Reply:
Sounds good, I can switch to that, it would avoid adding a user-facing parameter which would be good! Thanks for the suggestion!

I think calling attribute from different threads on the same attribution object isn't broadly supported by our methods, since we have other cases where we store temp info such as hook lists as an object attribute, which would have the same issue. But I agree that approach is cleaner, will update to it, thanks!

Reply:
Thank you :) yes, that's a good point. Ideally we need to make those hook holders local variables as well. IG is in a more sophisticated shape and more likely to be used than others. We can keep this in mind and make necessary adjustments in other approaches as well.